community/modules/scheduler/schedmd-slurm-gcp-v6-controller/slurm_files.tf

```diff
@@ -0,0 +1,129 @@
+# Copyright 2023 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+# BUCKET
+
+locals {
+  synt_suffix       = substr(md5("${var.project_id}${var.deployment_name}"), 0, 5)
+  synth_bucket_name = "${local.slurm_cluster_name}${local.synt_suffix}"
+
+  bucket_name = var.create_bucket ? module.bucket[0].name : var.bucket_name
+}
+
+module "bucket" {
+  source  = "terraform-google-modules/cloud-storage/google"
+  version = "~> 3.0"
+
+  count = var.create_bucket ? 1 : 0
+
+  location   = var.region
+  names      = [local.synth_bucket_name]
+  prefix     = "slurm"
+  project_id = var.project_id
+
+  force_destroy = {
+    (local.synth_bucket_name) = true
+  }
+
+  labels = {
+    slurm_cluster_name = local.slurm_cluster_name
+  }
+}
+
+# BUCKET IAMs
+locals {
+  controller_sa  = toset(flatten([for x in module.slurm_controller_template : x.service_account]))
+  compute_sa     = toset(flatten([for x in module.slurm_nodeset_template : x.service_account]))
+  compute_tpu_sa = toset(flatten([for x in module.slurm_nodeset_tpu : x.service_account]))
+  login_sa       = toset(flatten([for x in module.slurm_login_template : x.service_account]))
+
+  viewers = toset(flatten([
+    formatlist("serviceAccount:%s", [for x in local.controller_sa : x.email]),
+    formatlist("serviceAccount:%s", [for x in local.compute_sa : x.email]),
+    formatlist("serviceAccount:%s", [for x in local.compute_tpu_sa : x.email]),
+    formatlist("serviceAccount:%s", [for x in local.login_sa : x.email]),
+  ]))
+}
+
+
+resource "google_storage_bucket_iam_binding" "viewers" {
+  bucket  = local.bucket_name
+  role    = "roles/storage.objectViewer"
+  members = compact(local.viewers)
+}
+
+resource "google_storage_bucket_iam_binding" "legacy_readers" {
+  bucket  = local.bucket_name
+  role    = "roles/storage.legacyBucketReader"
+  members = compact(local.viewers)
+}
+
+# SLURM FILES
+locals {
+  ghpc_startup_script_controller = [{
+    filename = "ghpc_startup.sh"
```

Comment:
I think assumptions break down if I have multiple clusters using the same bucket. Very plausible now that we allow user to provide bucket. Could we add cluster name into the file names?

Reply:
By default files will be placed into separate directory with `cluster_name` in it.
Unless user specified both `bucket_name` and `bucket_dir` no collision should happen, 
otherwise we should let them to shoot in their foot since they tried so hard.